src/transformers/trainer_seq2seq.py

```diff
@@ -223,6 +224,9 @@ def predict(
             gen_kwargs["max_length"] = self.args.generation_max_length
         if gen_kwargs.get("num_beams") is None and self.args.generation_num_beams is not None:
             gen_kwargs["num_beams"] = self.args.generation_num_beams
+        # We don't want to drop samples if `num_return_sequences` > 1
+        if "generation_config" in gen_kwargs and gen_kwargs["generation_config"].num_return_sequences > 1:
+            self.gather_function = self.accelerator.gather
```

Comment:
Shouldn't this then just be `gather` for all cases here too?

Reply:
Yes, I did it for `evaluate` but not predict, fixing